File: Quiz-server/src/main/java/com/qucat/quiz/repositories/entities/AchievementCondition.java
package com.qucat.quiz.repositories.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@Builder
@AllArgsConstructor
public class AchievementCondition {
    @EqualsAndHashCode.Exclude private int id;
    private ConditionOperator operator;
    private int value;
    private int achievementId;
    private int characteristicId;
    @EqualsAndHashCode.Exclude private AchievementCharacteristic characteristic;
}

File: Quiz-web/src/app/modules/core/models/achievementCondition.ts
export class AchievementCondition {

  characteristicId: number;
  value: number;
  operator: ConditionOperator;

}

export class AchievementCharacteristic {

  id: number;
  name: string;

}

export enum ConditionOperator {
  LESS="<",
  EQUALS="=",
  GREATER=">"
}
